extensions/grpc/deployment/src/main/java/io/quarkus/grpc/deployment/GrpcServerProcessor.java

```diff
@@ -109,13 +110,13 @@ void processGeneratedBeans(CombinedIndexBuildItem index, BuildProducer<Annotatio
             BuildProducer<DelegatingGrpcBeanBuildItem> delegatingBeans) {
 
         // generated bean class -> blocking methods
-        Map<DotName, Set<MethodInfo>> generatedBeans = new HashMap<>();
+        Map<DotName, Set<String>> generatedBeans = new HashMap<>();
         String[] excludedPackages = { "grpc.health.v1", "io.grpc.reflection" };
 
         // We need to transform the generated bean and register a bindable service if:
         // 1. there is a user-defined bean that implements the generated interface (injected delegate)
         // 2. there is no user-defined bean that extends the relevant impl bases (both mutiny and regular)
-        for (ClassInfo generatedBean : index.getIndex().getKnownDirectImplementors(GrpcDotNames.MUTINY_BEAN)) {
+        for (ClassInfo generatedBean : index.getIndex().getAllKnownImplementors(GrpcDotNames.MUTINY_BEAN)) {
```

Comment:
I don't think that this change makes sense. This build step should only process all direct implementors of `MutinyBean`, i.e. the generated beans.

Reply:
Think it makes sense to allow inheritance and have multiple implementations (`@Default` and such).

Something like this:
```
@GrpcService
@Default
class MyServiceForProd extends MyServiceBaseImpl {
  protected doSomeWork(...) {
    // ...
  }
}

@GrpcService
@Dependent
@SomeAnnotationForSelection(DeploymentType.DEV)
class MyServiceForDev extends MyServiceBaseImpl {
  protected doSomeWork(...) {
    // ...
  }
}

class MyServiceBaseImpl extends MyServiceBaseImpl {
  @Blocking
  public Uni<...> foo(SomeRequest req) {
    // common code
    getBackend().doSomeWork(...);
    // common code
  }

  protected abstract doSomeWork(...);
}
```

Reply:
But `index.getIndex().getAllKnownImplementors(GrpcDotNames.MUTINY_BEAN)` would not return any of these classes. It should return the beans generated by quarkus.

Reply:
Ohhh!! I see.
Weird, the test cases work as expected though.

Reply:
Ah - got it now. Yea - that change doesn't make sense.